### PR TITLE
It2 season statistics

### PR DESCRIPTION
**Pull request by drewwilliams5280**
This is the method that takes 9 minutes and 42 seconds to run, do not merge yet! Just pull requesting for you guys to take a look at and see if you notice anything that might help. Nested each loops are definitely what's expensive in here.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -11,4 +11,53 @@ def initialize(data)
...
   end
 
+  def games_coached(season_id)
+    coaches = []
+    season = CSV.read(@data[:games], headers: true, header_converters: :symbol).find_all {|row| row[:season] == season_id}
+    games = CSV.read(@data[:game_teams], headers: true, header_converters: :symbol)
+    season.each do |season|
+      games.each do |game|
+        if season[:game_id] == game[:game_id]
+          coaches << game[:head_coach]
```
Comment (drewwilliams5280) [inline lib/stat_tracker.rb]: This code works, but is what I referenced as being expensive. It's essentially taking one line of the season CSV file, scanning every single line of the games file, then moving on to the second line of the season CSV file and scanning every single line, and so on. And considering there are like 15,000 rows of data, this is definitely not how to do it. Working on solution now.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -11,4 +11,53 @@ def initialize(data)
...
+  end
+
+  def games_won(season_id)
+    winning_coaches = []
+    season = CSV.read(@data[:games], headers: true, header_converters: :symbol).find_all {|row| row[:season] == season_id}
+    games = CSV.read(@data[:game_teams], headers: true, header_converters: :symbol)
+    season.each do |season|
+      games.each do |game|
+        if season[:game_id] == game[:game_id] && game[:result] == "WIN"
+          winning_coaches << game[:head_coach]
```
Comment (drewwilliams5280) [inline lib/stat_tracker.rb]: Same thing here essentially.

File: lib/stat_tracker.rb (hunk 1 above)
Comment (TravisGM92) [reply lib/stat_tracker.rb]: Yeah so this is what I mentioned earlier in Slack (and no worries if you missed it). If you put your CSV.read methods inside initialize, it only has to do it once. That's what I learned from your mentor, so I tried that on mine and it works.

File: test/stat_tracker_test.rb [hunk 3]
```diff
@@ -2,10 +2,89 @@
...
+      locations = {
+        games: game_path,
+        teams: team_path,
+        game_teams: game_teams_path
+        }
+      stat_tracker = StatTracker.from_csv(locations)
+      assert_equal "Claude Noel", stat_tracker.winningest_coach("20132014")
+    end
+
+  end
```
Comment (TravisGM92) [inline test/stat_tracker_test.rb:54]: Tests look good! Thanks for doing all of that

File: lib/stat_tracker.rb (hunk 1 above)
Comment (drewwilliams5280) [reply lib/stat_tracker.rb]: I have it initializing in self with .from_csv. He told me that was the way to go, but I'll see if putting it in with initialize speeds up the process further!

**Comment by TravisGM92** [comment]
We will be reorganizing our code so that there is more uniformity. It would be better if we all had similar looking ways to access our CSV files.

**Pull request closed without merging**